stdlib/public/Concurrency/Task.swift

```diff
@@ -15,22 +15,39 @@ import Swift
 
 // ==== Task -------------------------------------------------------------------
 
-/// An asynchronous task (just "Task" hereafter) is the analogue of a thread for
-/// asynchronous functions. All asynchronous functions run as part of some task.
+/// A unit of asynchronous work.
 ///
-/// A task can only be interacted with by code running "in" the task,
-/// by invoking the appropriate context sensitive static functions which operate
-/// on the "current" task. Because all such functions are `async` they can only
+/// All asynchronous functions run as part of some task.
+///
+/// Only code that's running as part of the task can interact with that task,
+/// by invoking the appropriate context-sensitive static functions which operate
+/// on the current task. Because all such functions are asynchronous, they can only
 /// be invoked as part of an existing task, and therefore are guaranteed to be
 /// effective.
+/// ◊TR: What do we mean by "effective" above?
 ///
 /// A task's execution can be seen as a series of periods where the task was
-/// running. Each such period ends at a suspension point or -- finally -- the
+/// running. Each such period ends at a suspension point or the
 /// completion of the task.
 ///
 /// These partial periods towards the task's completion are `PartialAsyncTask`.
-/// Partial tasks are generally not interacted with by end-users directly,
-/// unless implementing a scheduler.
+/// Unless you're implementing a scheduler,
+/// you don't generally interacted with partial tasks directly.
+///
+/// Task Cancellation
+/// =================
+///
+/// ◊ a task checks its own cancellation status
+/// ◊ you might need to check cancellation repeatedly for long-running work
+/// ◊ the canonical response to cancelation is to throw `Task.CancellationError`
```

Comment:
Yes, all these are correct.

The shortcut to "check and throw the cancellation error" is `try Task.checkCancellation()` but one might implement it just with `Task.isCancelled` and throwing